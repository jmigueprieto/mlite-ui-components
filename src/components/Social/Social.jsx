import React from "react"
import styles from "../../styles/styles.module.scss"

import ContactIcon from "../../assets/svg/contact.svg"
import FacebookIcon from "../../assets/svg/facebook.svg"
import GithubIcon from "../../assets/svg/github.svg"
import InstagramIcon from "../../assets/svg/instagram.svg"
import LinkedinIcon from "../../assets/svg/linkedin.svg"
import TwitterIcon from "../../assets/svg/twitter.svg"

const Social = ({ type, href, target }) => {
  const attrs = {
    contact: { src: ContactIcon, alt: "Contact" },
    facebook: { src: FacebookIcon, alt: "Facebook" },
    github: { src: GithubIcon, alt: "Github" },
    instagram: { src: InstagramIcon, alt: "Instagram" },
    linkedin: { src: LinkedinIcon, alt: "LinkedIn" },
    twitter: { src: TwitterIcon, alt: "Twitter" }
  }[type]

  if (!attrs) {
    throw Error(`Unknown type '${type}' of social icon`)
  }

  const img = <img className={styles.icon} src={attrs.src} alt={attrs.alt} />

  return (
    <a href={href} {...(target ? { target: target } : {})}>
      {img}
    </a>
  )
}

export default Social
