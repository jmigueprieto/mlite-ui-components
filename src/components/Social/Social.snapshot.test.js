import React from "react";
import renderer from "react-test-renderer";
import Social from "./Social";

describe("<Layout />", () => {
  it("renders correctly with type contact", () => {
    const tree = renderer
      .create(<Social type="contact" href="mailto:test@mprieto.me" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with type facebook", () => {
    const tree = renderer
      .create(
        <Social type="facebook" href="https://facebook.com/jmigueprieto" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with type github", () => {
    const tree = renderer
      .create(<Social type="github" href="https://github.com/jmigueprieto" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with type instagram", () => {
    const tree = renderer
      .create(
        <Social type="instagram" href="https://instagram.com/jmigueprieto" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with type linkedin", () => {
    const tree = renderer
      .create(
        <Social type="linkedin" href="https://linkedin.com/in/jmigueprieto" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with type twitter", () => {
    const tree = renderer
      .create(
        <Social type="twitter" href="https://twitter.com/in/miguelprieto" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
