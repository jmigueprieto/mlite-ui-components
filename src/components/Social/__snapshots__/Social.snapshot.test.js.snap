// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> renders correctly with type contact 1`] = `
<a
  href="mailto:test@mprieto.me"
>
  <svg
    className="icon"
    version="1.1"
    viewBox="0 0 16 16"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <path
      d="M8,10c-0.266,0-0.5-0.094-1-0.336L0,6v7c0,0.55,0.45,1,1,1h14c0.55,0,1-0.45,1-1V6L9,9.664C8.5,9.906,8.266,10,8,10z M15,2  H1C0.45,2,0,2.45,0,3v0.758l8,4.205l8-4.205V3C16,2.45,15.55,2,15,2z"
      fill="#ccc"
    />
  </svg>
</a>
`;

exports[`<Layout /> renders correctly with type facebook 1`] = `
<a
  href="https://facebook.com/jmigueprieto"
>
  <svg
    className="icon"
    version="1.1"
    viewBox="0 0 56.693 56.693"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <path
      d="M28.347,5.157c-13.6,0-24.625,11.027-24.625,24.625c0,13.6,11.025,24.623,24.625,24.623c13.6,0,24.625-11.023,24.625-24.623  C52.972,16.184,41.946,5.157,28.347,5.157z M34.864,29.679h-4.264c0,6.814,0,15.207,0,15.207h-6.32c0,0,0-8.307,0-15.207h-3.006  V24.31h3.006v-3.479c0-2.49,1.182-6.377,6.379-6.377l4.68,0.018v5.215c0,0-2.846,0-3.398,0c-0.555,0-1.34,0.277-1.34,1.461v3.163  h4.818L34.864,29.679z"
      fill="#ccc"
    />
  </svg>
</a>
`;

exports[`<Layout /> renders correctly with type github 1`] = `
<a
  href="https://github.com/jmigueprieto"
>
  <svg
    className="icon"
    version="1.1"
    viewBox="0 0 1024 1024"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 0C3.58 0 0 3.58 0 8C0 11.54 2.29 14.53 5.47 15.59C5.87 15.66 6.02 15.42 6.02 15.21C6.02 15.02 6.01 14.39 6.01 13.72C4 14.09 3.48 13.23 3.32 12.78C3.23 12.55 2.84 11.84 2.5 11.65C2.22 11.5 1.82 11.13 2.49 11.12C3.12 11.11 3.57 11.7 3.72 11.94C4.44 13.15 5.59 12.81 6.05 12.6C6.12 12.08 6.33 11.73 6.56 11.53C4.78 11.33 2.92 10.64 2.92 7.58C2.92 6.71 3.23 5.99 3.74 5.43C3.66 5.23 3.38 4.41 3.82 3.31C3.82 3.31 4.49 3.1 6.02 4.13C6.66 3.95 7.34 3.86 8.02 3.86C8.7 3.86 9.38 3.95 10.02 4.13C11.55 3.09 12.22 3.31 12.22 3.31C12.66 4.41 12.38 5.23 12.3 5.43C12.81 5.99 13.12 6.7 13.12 7.58C13.12 10.65 11.25 11.33 9.47 11.53C9.76 11.78 10.01 12.26 10.01 13.01C10.01 14.08 10 14.94 10 15.21C10 15.42 10.15 15.67 10.55 15.59C13.71 14.53 16 11.53 16 8C16 3.58 12.42 0 8 0Z"
      fill="#ccc"
      transform="scale(64)"
    />
  </svg>
</a>
`;

exports[`<Layout /> renders correctly with type instagram 1`] = `
<a
  href="https://instagram.com/jmigueprieto"
>
  <svg
    className="icon"
    version="1.1"
    viewBox="0 0 56.693 56.693"
  >
    <path
      d="M43.414,4.831H13c-5.283,0-9.581,4.297-9.581,9.58v10.12v20.295c0,5.283,4.298,9.58,9.581,9.58h30.415  c5.283,0,9.58-4.297,9.58-9.58V24.53v-10.12C52.994,9.127,48.697,4.831,43.414,4.831z M46.162,10.544l1.096-0.004v1.091v7.312  l-8.375,0.027l-0.029-8.402L46.162,10.544z M21.131,24.53c1.588-2.197,4.164-3.638,7.076-3.638s5.488,1.441,7.074,3.638  c1.033,1.434,1.652,3.188,1.652,5.088c0,4.811-3.918,8.725-8.727,8.725c-4.812,0-8.726-3.914-8.726-8.725  C19.481,27.718,20.098,25.964,21.131,24.53z M48.164,44.825c0,2.621-2.131,4.75-4.75,4.75H13c-2.62,0-4.751-2.129-4.751-4.75V24.53  h7.4c-0.639,1.572-0.998,3.289-0.998,5.088c0,7.473,6.08,13.557,13.556,13.557c7.475,0,13.555-6.084,13.555-13.557  c0-1.799-0.361-3.516-1-5.088h7.402V44.825z"
      fill="#ccc"
    />
  </svg>
</a>
`;

exports[`<Layout /> renders correctly with type linkedin 1`] = `
<a
  href="https://linkedin.com/in/jmigueprieto"
>
  <svg
    className="icon"
    version="1.1"
    viewBox="0 0 56.693 56.693"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <g
      fill="#ccc"
    >
      <path
        d="M30.071,27.101v-0.077c-0.016,0.026-0.033,0.052-0.05,0.077H30.071z"
      />
      <path
        d="M49.265,4.667H7.145c-2.016,0-3.651,1.596-3.651,3.563v42.613c0,1.966,1.635,3.562,3.651,3.562h42.12   c2.019,0,3.654-1.597,3.654-3.562V8.23C52.919,6.262,51.283,4.667,49.265,4.667z M18.475,46.304h-7.465V23.845h7.465V46.304z    M14.743,20.777h-0.05c-2.504,0-4.124-1.725-4.124-3.88c0-2.203,1.67-3.88,4.223-3.88c2.554,0,4.125,1.677,4.175,3.88   C18.967,19.052,17.345,20.777,14.743,20.777z M45.394,46.304h-7.465V34.286c0-3.018-1.08-5.078-3.781-5.078   c-2.062,0-3.29,1.389-3.831,2.731c-0.197,0.479-0.245,1.149-0.245,1.821v12.543h-7.465c0,0,0.098-20.354,0-22.459h7.465v3.179   c0.992-1.53,2.766-3.709,6.729-3.709c4.911,0,8.594,3.211,8.594,10.11V46.304z"
      />
    </g>
  </svg>
</a>
`;

exports[`<Layout /> renders correctly with type twitter 1`] = `
<a
  href="https://twitter.com/in/miguelprieto"
>
  <svg
    className="icon"
    version="1.1"
    viewBox="0 0 56.693 56.693"
  >
    <path
      d="M52.837,15.065c-1.811,0.805-3.76,1.348-5.805,1.591c2.088-1.25,3.689-3.23,4.444-5.592c-1.953,1.159-4.115,2-6.418,2.454  c-1.843-1.964-4.47-3.192-7.377-3.192c-5.581,0-10.106,4.525-10.106,10.107c0,0.791,0.089,1.562,0.262,2.303  c-8.4-0.422-15.848-4.445-20.833-10.56c-0.87,1.492-1.368,3.228-1.368,5.082c0,3.506,1.784,6.6,4.496,8.412  c-1.656-0.053-3.215-0.508-4.578-1.265c-0.001,0.042-0.001,0.085-0.001,0.128c0,4.896,3.484,8.98,8.108,9.91  c-0.848,0.23-1.741,0.354-2.663,0.354c-0.652,0-1.285-0.063-1.902-0.182c1.287,4.015,5.019,6.938,9.441,7.019  c-3.459,2.711-7.816,4.327-12.552,4.327c-0.815,0-1.62-0.048-2.411-0.142c4.474,2.869,9.786,4.541,15.493,4.541  c18.591,0,28.756-15.4,28.756-28.756c0-0.438-0.009-0.875-0.028-1.309C49.769,18.873,51.483,17.092,52.837,15.065z"
      fill="#ccc"
    />
  </svg>
</a>
`;
